src/libmongoc/tests/TestSuite.c

```diff
@@ -263,12 +251,8 @@ TestSuite_Add (TestSuite *suite, /* IN */
                const char *name, /* IN */
                TestFunc func)    /* IN */
 {
-   TestSuite_AddFull (suite,
-                      name,
-                      TestSuite_AddHelper,
-                      NULL,
-                      (void *) func,
-                      TestSuite_CheckDummy);
+   TestSuite_AddFullWithTestFn (
+      suite, name, TestSuite_AddHelper, NULL, func, TestSuite_CheckDummy);
```

Comment:
This and the below call within `TestSuite_AddLive` are broken. The `static` test context instance is only instantiated within the scope of this function, so each test that calls `TestSuite_Add()` will shared the static instance, causing only the final caller to actually have its test function registered. Each test that was defined using this function will instead that final test function instead of their own test function.

It may be enough to change `TestSuite_Add` and `TestSuite_AddLive` to also be macros, but that might just be kicking the can further down the road.

Reply:
Good spot! I had a feeling opting for this "simpler" approach might come back to bite me. It seems the principle of keeping scope (and lifetime) as limited to necessary scope as possible proves itself again here.

Thankfully I already explored and documented an alternative solution that does exactly that, which should directly address this issue you identified. I will document this issue on JIRA and post a corresponding PR shortly.

Reply:
PR posted: https://github.com/mongodb/mongo-c-driver/pull/865